lib/producer.ts

```diff
@@ -37,14 +37,25 @@ function discoverPlaceholder(
   binaryBuffer: Buffer,
   searchString: string,
   padder: string,
+  searchOffset: number = 0,
 ): Placeholder | NotFound {
   const placeholder = Buffer.from(searchString);
-  const position = binaryBuffer.indexOf(placeholder);
+  const position = binaryBuffer.indexOf(placeholder, searchOffset);
 
   if (position === -1) {
     return { notFound: true };
   }
 
+  // reject source code literal if it occurs first in binary
```

Comment:
add a comment here that reference to this PR and also explains when this happens (like the text you put in the PR)